clippy_lints/src/unflagged_test_modules.rs

```diff
@@ -0,0 +1,73 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use rustc_hir::*;
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use rustc_span::{symbol::sym, BytePos};
+
+declare_clippy_lint! {
+    /// ### What it does
+    ///
+    /// Triggers when a testing module (that contains "test" or "tests" in its name) isn't flagged with the `#[cfg(test)]` attribute.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// The attribute `#[cfg(test)]` is used to tell Rust to compile and run the test code only when you run `cargo test` and not when you run `cargo  build`. This saves compile time and space in the resulting compiled artifact because tests are not included. So not using `#[cfg(test)]` for tests is both a waste of time and space.
+    ///
+    /// ### Example
+    /// ```rust
+    /// mod my_cool_tests {
+    /// 	// [...]
+    /// }
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// #[cfg(test)]
+    /// mod my_cool_tests {
+    /// 	// [...]
+    /// }
+    /// ```
+    #[clippy::version = "1.70.0"]
+    pub UNFLAGGED_TEST_MODULES,
+    perf,
```

Comment:
This is definitely not a perf lint

Reply:
What does perf mean then? I thought it was performance.

Reply:
The performance of the compiled code, not the performance of the compilation step itself.

Reply:
Reverted